### PR TITLE
[FIX] html_editor: add direction config

### PR DESCRIPTION
**Pull request by Mtaylorr**

### CODE REVIEW

File: addons/html_editor/static/tests/html_field.test.js [hunk 1]
```diff
@@ -1531,3 +1532,57 @@ describe("sandbox", () => {
         );
     });
 });
+
+describe("direction config", () => {
+    test("ltr direction", async () => {
+        onRpc("partner", "web_save", ({ args }) => {
+            expect(args[1].txt).toBe("<div></div>");
+            expect.step("web_save");
+        });
```
Comment (FrancoisGe) [inline addons/html_editor/static/tests/html_field.test.js]: you don t need it

File: addons/html_editor/static/tests/html_field.test.js [hunk 2]
```diff
@@ -1531,3 +1532,57 @@ describe("sandbox", () => {
...
+            resModel: "partner",
+            arch: `
+            <form>
+                <field name="txt" widget="html"/>
+            </form>`,
+        });
+        expect(".odoo-editor-editable").toHaveAttribute("dir", "ltr");
+        const node = queryOne(".odoo-editor-editable p");
+        setSelection({ anchorNode: node, anchorOffset: 0, focusNode: node, focusOffset: 1 });
+        switchDirection(htmlEditor);
```
Comment (FrancoisGe) [inline addons/html_editor/static/tests/html_field.test.js]: use a real user flow. Open command palette and select switch direction (insertText(htmlEditor, "/switch") + press("enter")

File: addons/html_editor/static/tests/html_field.test.js [hunk 3]
```diff
@@ -1531,3 +1532,57 @@ describe("sandbox", () => {
...
+        expect(".odoo-editor-editable p").toHaveAttribute("dir", "rtl");
+    });
+
+    test("rtl direction", async () => {
+        onRpc("partner", "web_save", ({ args }) => {
```
Comment (FrancoisGe) [inline addons/html_editor/static/tests/html_field.test.js]: same

File: addons/html_editor/static/tests/html_field.test.js [hunk 4]
```diff
@@ -1531,3 +1532,57 @@ describe("sandbox", () => {
...
+            resId: 1,
+            resModel: "partner",
+            arch: `
+            <form>
+                <field name="txt" widget="html"/>
+            </form>`,
+        });
+        expect(".odoo-editor-editable").toHaveAttribute("dir", "rtl");
+        const node = queryOne(".odoo-editor-editable p");
+        setSelection({ anchorNode: node, anchorOffset: 0, focusNode: node, focusOffset: 1 });
```
Comment (FrancoisGe) [inline addons/html_editor/static/tests/html_field.test.js]: you can use setSelectionInHtmlField

File: addons/html_editor/static/tests/html_field.test.js [hunk 5]
```diff
@@ -1531,3 +1532,57 @@ describe("sandbox", () => {
...
+        switchDirection(htmlEditor);
```
Comment (FrancoisGe) [inline addons/html_editor/static/tests/html_field.test.js]: same use powerbox

File: addons/html_editor/static/tests/html_field.test.js (hunk 4 above)
Comment (Mtaylorr) [reply addons/html_editor/static/tests/html_field.test.js]: @FrancoisGe I had a problem here if I put the selection in the `p` element, I added a comment about it in the code

File: addons/html_editor/static/tests/html_field.test.js [hunk 6]
```diff
@@ -1531,3 +1532,63 @@ describe("sandbox", () => {
...
+        expect(".odoo-editor-editable").toHaveAttribute("dir", "rtl");
+        // If we force selection in [P,0] it doesn't work because
+        // anchorNode is p so we filter it
+        // getSelectedNodes returns nothing since the textNode isn't fully selected
+        const node = queryOne(".odoo-editor-editable p");
+        setSelection({ anchorNode: node.firstChild, anchorOffset: 0 });
+        insertText(htmlEditor, "/Switchdirection");
+        await animationFrame();
+        expect(queryAllTexts(".o-we-command-name")).toEqual(["Switch direction"]);
+        expect(".o-we-powerbox").toHaveCount(1);
```
Comment (FrancoisGe) [inline addons/html_editor/static/tests/html_field.test.js]: you don need this line by default you check it with the previous

**Pull request merged**